### PR TITLE
Deployment details no longer accessed from global context

### PR DESCRIPTION
**Pull request by robfletcher**
This ensures deployment details are retrieved from upstream stages, trigger or parent pipeline without using global context

### CODE REVIEW

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy [hunk 1]
```diff
@@ -224,14 +224,16 @@ class PackageInfo {
...
-      artifactMatch(it.context.buildInfo?.artifacts as List<Map<String, String>>, packageFilePattern)
+      artifactMatch(it.outputs.buildInfo?.artifacts as List<Map<String, String>>, packageFilePattern)
     }
-    return upstreamStage ? upstreamStage.context.buildInfo as Map : null
+    return upstreamStage ? upstreamStage.outputs.buildInfo as Map : null
   }
 
-  private static boolean artifactMatch(List<Map<String, String>> artifacts, Pattern pattern) {
+  private
+  static boolean artifactMatch(List<Map<String, String>> artifacts, Pattern pattern) {
```
Comment (ajordens) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy]: mr line breaky break

**Comment by ajordens** [comment]
Are all the stages putting the right bits into `outputs`? Guessing you had changed that previously?

**Comment by ajordens** [comment]
Worth sanity checking that deployment strategies still work with both a bake and find image stage in the parent stage (verify both independently).

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy (hunk 1 above)
Comment (robfletcher) [reply orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy]: Ugh.

**Pull request merged**